vitessce/wrappers.py

```diff
@@ -461,14 +466,341 @@ def __init__(self, loom, **kwargs):
+                cluster_df = in_clusters_df.loc[in_clusters_df["cluster"] == cluster_id]
+                cluster_cell_ids = cluster_df.index.values.tolist()
+                cluster_num_cells = len(cluster_cell_ids)
+                cluster_cells_tf = (in_barcodes_df[0].isin(cluster_cell_ids)).values
+
+                # Get the rows of the data matrix corresponding to the cells in this cluster.
+                cluster_cell_by_bin_mtx = chr_mtx[cluster_cells_tf,:]
+                # Take the sum of this cluster along the cells axis.
+                cluster_profile = cluster_cell_by_bin_mtx.sum(axis=0)
+
+                # Fill in the data for this cluster and chromosome at each resolution.
+                for resolution, resolution_exp in zip(resolutions, resolution_exps):
+                    arr_len = math.ceil(chr_len / resolution)
+                    chr_shape = (num_clusters, arr_len)
+
+                    # Pad the array of values with zeros if necessary before reshaping.
+                    values = cluster_profile
+                    padding_len = resolution_exp - (values.shape[0] % resolution_exp)
+                    if values.shape[0] % resolution_exp > 0:
+                        values = np.concatenate((values, np.zeros((padding_len,))))
+                    num_tiles = chr_shape[1]
+                    # Reshape to be able to sum every `resolution_exp` number of values.
+                    arr = np.reshape(values, (-1, resolution_exp)).sum(axis=-1)
+
+                    padding_len = arr_len - arr.shape[0]
+                    if padding_len > 0:
+                        arr = np.concatenate((arr, np.zeros((padding_len,))))
+                    # Set the array in the Zarr store.
+                    chromosomes_group[chr_name][str(resolution)][cluster_index,:] = arr
+       
+        # out_f.attrs should contain the properties required for HiGlass's "tileset_info" requests.
+        out_f.attrs['row_infos'] = [
+            { "cluster": cluster_id }
+            for cluster_id in cluster_ids
+        ]
+        out_f.attrs['resolutions'] = sorted(resolutions, reverse=True)
+        out_f.attrs['shape'] = [ num_clusters, 256 ]
+        out_f.attrs['name'] = "SnapTools"
+        out_f.attrs['coordSystem'] = "hg38"
+        
+        # https://github.com/zarr-developers/zarr-specs/issues/50
+        out_f.attrs['multiscales'] = [
+            {
+                "version": "0.1",
+                "name": chr_name,
+                "datasets": [
+                    { "path": f"chromosomes/{chr_name}/{resolution}" }
+                    for resolution in sorted(resolutions, reverse=True)
+                ],
+                "type": "zarr-multivec",
+                "metadata": {
+                    "chromoffset": int(chrom_name_to_cumsum[chr_name]),
+                    "chromsize": int(chr_len),
+                }
+            }
+            for (chr_name, chr_len) in list(zip(chromosomes, chroms_length_arr))
+        ]
```

Comment:
I think this is what I'm referring to re: factoring out.  I'm not going to block on it because I don't know enough either way, but ti seems that this is the sort of thing that could be reused.

Reply:
I see, yes I can do that